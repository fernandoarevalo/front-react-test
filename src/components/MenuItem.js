import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const MenuButton = styled.button`
  border: none;
  width: 100%;

  &,
  :focus {
    background-color: transparent;
  }
`;

export default function MenuItem({ airlineObj, selected, setAirLine }) {
  return (
    <li
      role="tab"
      aria-selected={selected}
      className={`collection-item ${selected ? "active" : ""}`}
    >
      <MenuButton
        type="button"
        onClick={() => {
          setAirLine(airlineObj);
        }}
      >
        {airlineObj.name}
      </MenuButton>
    </li>
  );
}

/**
 * Option proptypes
 */
const Option = PropTypes.shape({
  id: PropTypes.string,
  name: PropTypes.string,
});

/**
 * PropsTypes
 */
MenuItem.propTypes = {
  airlineObj: Option,
  selected: PropTypes.bool,
  setAirLine: PropTypes.func,
};

/**
 * Default props
 */
MenuItem.defaultProps = {
  airlineObj: {},
  selected: false,
  setAirLine: null,
};
