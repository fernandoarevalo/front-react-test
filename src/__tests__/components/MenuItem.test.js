import React from "react";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import MenuItem from "components/MenuItem";
import AIRLINES from "utils/mockups/Airlines";

describe("<MenuItem />", () => {
  const setAirLine = jest.fn();

  afterEach(() => {
    cleanup();
  });

  test("Calls onClick prop when Menu item clicked", () => {
    render(
      <MenuItem
        key={AIRLINES[0].id}
        airlineObj={AIRLINES[0]}
        setAirLine={setAirLine}
      />
    );
    fireEvent.click(screen.getByRole("tab", { selected: false }));
  });

  test("Snapshot MenuItem component", () => {
    const { asFragment } = render(
      <MenuItem
        key={AIRLINES[0].id}
        airlineObj={AIRLINES[0]}
        setAirLine={setAirLine}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
