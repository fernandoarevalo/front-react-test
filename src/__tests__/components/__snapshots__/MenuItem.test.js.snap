// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> Snapshot MenuItem component 1`] = `
<DocumentFragment>
  <li
    aria-selected="false"
    class="collection-item "
    role="tab"
  >
    <button
      class="sc-bdfBwQ hGbOiT"
      type="button"
    >
      Vivaair
    </button>
  </li>
</DocumentFragment>
`;
